File: UsaloYa.API/Models/Sale.cs
using System;
using System.Collections.Generic;

namespace UsaloYa.API.Models;

public partial class Sale
{
    public int SaleId { get; set; }

    public int? CustomerId { get; set; }

    public DateTime SaleDate { get; set; }

    public string PaymentMethod { get; set; } = null!;

    public decimal Tax { get; set; }

    public string Status { get; set; } = null!;

    public decimal TotalSale { get; set; }

    public string? Notes { get; set; }

    public int UserId { get; set; }

    public int CompanyId { get; set; }

    public virtual ICollection<SaleDetail> SaleDetails { get; set; } = new List<SaleDetail>();

    public virtual User User { get; set; } = null!;

    public virtual Company Company { get; set; } = null!;
}

File: Guardian/src/app/dto/sale.ts
export interface SaleDetail {
    ProductId: number;
    Quantity: number;
    UnitPrice: number;
    TotalPrice: number;
    SaleId:number;
}
  
export interface Sale {
    id: number | undefined, /*Used for internal offline sync management */
    saleId: number,
    customerId: number | undefined;
    paymentMethod: string;
    tax: number;
    notes: string;
    userId: number;
    companyId: number;
    saleDetailsList: SaleDetail[];
}